qcodes/instrument_drivers/yokogawa/GS200.py

```diff
@@ -191,11 +191,10 @@ def __init__(self, name: str, address: str, terminator: str = "\n",
                            set_cmd=self._set_source_mode,
                            vals=Enum('VOLT', 'CURR'))
 
-        # When getting the mode internally in the driver, look up the mode as
-        # recorded by the _cached_mode property, instead of calling source
-        # _mode(). This will prevent frequent VISA calls to the instrument.
-        # Calling _set_source_mode will change the chased value.
-        self._cached_mode = "VOLT"
+        # The source_mode attribute of the instrument. The default mode is
+        # VOLT when the instrument is truned on. The mode can be switched
+        # between VOLT and CURR.
+        self.source_mode()
```

Comment:
Typo `truned on`

As it is this comment is not very useful. It should explain why we get the mode here. This is because at connect time it is what ever the user set it to the last time or voltage if it has not been set.

Reply:
The code from line 232 to 247 still assumes that we are in voltage mode when starting up. This is wrong. The snapshot_exlude needs to be set depending on the mode we are in.

Reply:
Same is true for voltage_range and current_range around line 204

Reply:
I think this is better:
This attribute returns the latest user-set mode (VOLT or CURR) upon initialization of the instrument. It returns VOLT as the default mode of the instrument if nothing is set.

Reply:
I would suggest something like

`we need to get the value of source_mode here as we cannot rely on the default value
as the user may have changed the value before we connect to the instrument (in a previous session or via the 
frontpanel)`

We do not need to document that this returns the value. That is obvious from how parameters work. We only need to document why we need to update it here